### PR TITLE
calls initialize method on TrellisHttpResource to setup root ACL

### PR DESCRIPTION
**Pull request by gregjan**
Uses Trellis code to initialize the Root container with a default ACL.

### CODE REVIEW

**Comment by gregjan** [comment]
Hang on, something is definitely wrong here.

**Comment by gregjan** [comment]
Had some code bleed over from my other branch. Should be fixed now.

**Comment by ajs6f** [comment]
We're good to merge?

File: webapp/src/main/java/edu/si/trellis/CassandraApplication.java [hunk 1]
```diff
@@ -107,7 +108,9 @@ private void addConfig(URL url) {
 
     @Override
     public Set<Object> getSingletons() {
-        return ImmutableSet.of(new TrellisHttpResource(services), new TrellisHttpFilter(), new TrellisWebDAV(services),
+    	TrellisHttpResource thr = new TrellisHttpResource(services);
+    	runAsync(() -> { thr.initialize(); });
```
Comment (ajs6f) [inline webapp/src/main/java/edu/si/trellis/CassandraApplication.java]: Can this just be `runAsync(thr::initialize);`?

**Comment by gregjan** [comment]
absolutely

**Comment by gregjan** [comment]
I'll fix it after I eat lunch.

**Comment by gregjan** [comment]
delayed, but I added that commit to make it cleaner..

**Comment by gregjan** [comment]
I think failure and fast, i.e. then not starting is the way to go. Without
the default root acl, but with the filter enabled, the repo is unusable
anyway. A restart of the app (or the docker machine) might then fix it, if
the db became available..
Deeper answer is that the call should only exist when webac is included in
wiring, but that's for later.

**Comment by ajs6f** [comment]
Right, but does this PR do what you just described? IIUC, the resulting exception will get swallowed, won't it? Why would the server threads (JAX-RS threads) see it? I think we need to take more positive action on a failure…

**Comment by acoburn** [comment]
As per the [solid WAC spec](https://github.com/solid/web-access-control-spec#acl-inheritance-algorithm), there `MUST` be an ACL in the root container. The spec doesn't define what happens if there isn't an ACL there, but the Trellis WebAC module is written to prevent user access.

However, if an authenticated user is an "Admin" user, then WebAC is bypassed. Admin users are defined in the [AgentAuthorizationFilter](https://www.trellisldp.org/docs/trellis/current/apidocs/org/trellisldp/http/AgentAuthorizationFilter.html) class, or via the `trellis.http.agent.adminusers` configuration key (comma-separated list of WebIDs). Either way, you'll want to be sure that the `AgentAuthorizationFilter` class gets injected into the JAX-RS runtime.

**Comment by ajs6f** [comment]
Okay, if the result of a failure in initialization is that the module will then prevent user access, I'm okay with that. We do need to check that the exception will get properly logged, or the administrators will never be able to figure out what is happening. I can do that later this evening-- I'll just alter the PR to intentionally throw an exception and make sure I can see it, or if we want to hurry, I can merge this now and test that later. @gregjan ?

**Comment by gregjan** [comment]
No need to hurry here. I can devote more time to the testbed
parameterization today.

**Comment by gregjan** [comment]
And in total agreement with regards to making failure prevent startup.
My Java 8 async chops are pretty non-existent, so thanks for taking
that.

**Comment by ajs6f** [comment]
Okay, here's what's happening: I realized on the train that doing CDI for this isn't that hard. I therefore have https://github.com/trellis-ldp/trellis/pull/405 to enable CDI injection for some of the WebDAV and WebAC gear, matched with https://github.com/gregjan/trellis-cassandra/pull/1, which is a PR against _this_ PR that takes advantage of that to just let CDI do its `@PostConstruct` thing just like it should.

What do you guys say?

**Comment by acoburn** [comment]
If this is what's needed, then I'm all in favor of the change.

**Comment by ajs6f** [comment]
Well, what @gregjan has here will certainly work, it's more that we want to go towards CDI anyway and the CDI version of this, to my mind, is cleaner in the sense that we are leaving wiring and init to the framework, which is actually the whole point of paying for CDI.

**Pull request merged**